src/rest/handlers/rest_edges.erl

```diff
@@ -185,120 +185,78 @@ modify_state(delete, From, To, _, _, _) ->
-                     end;
-                 % wait for all pids to report back and then look for errors
-                 F([], AllResults) ->
-                     Errors = lists:filtermap(fun
-                                                  ({'$pmap_error', Pid, Type, Reason, Stacktrace}) ->
-                                                      {true, {Pid, Type, Reason, Stacktrace}};
-                                                  (_) ->
-                                                      false
-                                              end, AllResults),
-
-                     case Errors of
-                         [] ->
-                             lists:all(fun(X) -> X end, AllResults);
-                         _ ->
-                             false
-                     end
-             end,
+    Gather = fun F(PendingPids = [_ | _], PidsOrResults) ->
+        receive {Pid, Result} ->
+            NewPidsOrResults = rest_utils:replace(Pid, Result, PidsOrResults),
+            F(lists:delete(Pid, PendingPids), NewPidsOrResults)
+        after 5000 ->
+            case lists:any(fun erlang:is_process_alive/1, PendingPids) of
+                true -> F(PendingPids, PidsOrResults);
+                false -> error({parallel_call_failed, {processes_dead, Pids}}) end
+            end;
+        F([], AllResults) ->
+            Errors =
+                lists:filtermap(
+                    fun({'$pmap_error', Pid, Type, Reason, Stacktrace}) ->
+                        {true, {Pid, Type, Reason, Stacktrace}};
+                        (_) -> false
+                    end, AllResults),
+
+            case Errors of
+                [] -> lists:all(fun(X) -> X end, AllResults);
+                _ -> false
+            end end,
     Gather(Pids, Pids).
 
 -spec modify_state_bulk(SrcZone :: binary(), DstZone :: binary(), Edges :: list(map()), Successive :: boolean(), OneZoneOperation :: boolean()) -> ok | {error, any()}.
 modify_state_bulk(_, _, _, false, true) ->
     ok;
 modify_state_bulk(SrcZone, DstZone, Edges, _, _)->
     Parent = self(),
-    Ref = erlang:make_ref(),
 
     Pids = lists:map(fun(#{from := FromName, to := ToName, permissions := Permissions, trace := _Trace}) ->
         spawn(fun() ->
-            Result = try
-                         graph:create_edge(gmm_utils:create_vertex_id(SrcZone,  FromName), gmm_utils:create_vertex_id(DstZone, ToName), Permissions)
-                    catch Type:Reason:Stacktrace ->
-                        {'$pmap_error', self(), Type, Reason, Stacktrace}
-                    end,
-                    Parent ! {Ref, self(), Result}
-                end)
+            Result = try graph:create_edge(gmm_utils:create_vertex_id(SrcZone,  FromName), gmm_utils:create_vertex_id(DstZone, ToName), Permissions)
+                     catch Type:Reason:Stacktrace -> {'$pmap_error', self(), Type, Reason, Stacktrace} end,
+            Parent ! {self(), Result} end)
             end, Edges),
```

Comment:
Again, make sure indentations are matching, so that it's easy to see the logic of this code